model/src/market/agreement_event.rs

```diff
@@ -0,0 +1,58 @@
+/*
+ * Yagna Market API
+ *
+ * The version of the OpenAPI document: 1.6.1
+ *
+ * Generated by: https://openapi-generator.tech
+ */
+
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use crate::market::Reason;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(tag = "eventtype")]
+pub enum AgreementOperationEvent {
```

Comment:
If agent is both requestor and provider he can't distinguish between AgreementTerminated events

Reply:
This is the agent's responsibility to distinguish this event. I can imagine the agent being provider and requestor at the same time, but even then the agent can distinguish using agreement contents (there is Offer id and Demand id, which agent can compare to his own)

Reply:
Offer and Demand id is the same on provider and requestor. He can't distiguish

Reply:
It is only for the case Provider and Requestor are using the same identity and same (or no) `appSessionId`.